frontends/p4/strengthReduction.cpp

```diff
@@ -288,6 +288,12 @@ const IR::Node* DoStrengthReduction::postorder(IR::Slice* expr) {
             return new IR::Concat(expr->type, expr,
                     new IR::Constant(IR::Type_Bits::get(-(lo + shift_amt)), 0)); } }
 
+    while (auto cast = expr->e0->to<IR::Cast>()) {
+        if (size_t(expr->getH()) < cast->expr->type->width_bits()) {
```

Comment:
since you are doing casts you may perhaps want to also handle unsigned casts by converting them into either slices or concatenations

Reply:
Concatenation may well be more expensive than casts, so converting casts to concatenations should only be done for targets that want that, and not in the target-independent front end code.

Reply:
That's fine with me.